### PR TITLE
support building trust tore from a ca cert file

### PR DESCRIPTION
**Pull request by smrutilal2**
1. Enabled Junit test cases in the maven build.
2. Fixed existing test failure.
3. Added support to build trust store from a ca cert file.

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

File: libs/java/cert_refresher/src/main/java/com/oath/auth/TrustStore.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import java.security.KeyStore;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * A trust store built from a jks file by default. Or from the keystore provided by the
```
Comment (SeanPMiller) [inline libs/java/cert_refresher/src/main/java/com/oath/auth/TrustStore.java]: Incomplete doc?

File: libs/java/cert_refresher/src/main/java/com/oath/auth/KeyStoreProvider.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * limitations under the License.
+ */
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+public interface KeyStoreProvider {
```
Comment (SeanPMiller) [inline libs/java/cert_refresher/src/main/java/com/oath/auth/KeyStoreProvider.java]: Please write docs for interface and `provide()` method.

File: libs/java/cert_refresher/src/test/java/com/oath/auth/SocketTest.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ private void runPingServer(int port, KeyRefresher keyRefresher) throws Exception
         }).start();
     }
 
-    @Test
+//    @Test
```
Comment (SeanPMiller) [inline libs/java/cert_refresher/src/test/java/com/oath/auth/SocketTest.java]: Should this test be reenabled?

File: libs/java/cert_refresher/src/test/resources/gdpr.aws.core.key.pem [hunk 4]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (SeanPMiller) [inline libs/java/cert_refresher/src/test/resources/gdpr.aws.core.key.pem]: Yikes! Where did this come from?

Comment (smrutilal2) [reply libs/java/cert_refresher/src/test/resources/gdpr.aws.core.key.pem]: This is a test key.

File: libs/java/cert_refresher/src/test/java/com/oath/auth/SocketTest.java (hunk 3 above)
Comment (smrutilal2) [reply libs/java/cert_refresher/src/test/java/com/oath/auth/SocketTest.java]: FYI.  Tests were not running initially.  This particular test like many others was not complete and need little more time. But yes, we should fix and enable it.

**Comment by havetisyan** [comment]
@smrutilal2 Can you please provide some documentation as Sean pointed out.

Also, I noticed that the PR is changing the KeyRefresh class definition. Was this not already used by other teams or other teams were only using it through Utils.generateKeyRefresher to access the key refresher. @mmannerm can you please confirm that you're not affected by this change. thanks.

File: libs/java/cert_refresher/src/main/java/com/oath/auth/Utils.java [hunk 5]
```diff
@@ -86,9 +78,20 @@ public static KeyStore getKeyStore(final String jksFilePath) throws Exception {
      *  It requires that the proxies are created which are then stored in the KeyRefresher
      */
     public static KeyRefresher generateKeyRefresher(final String trustStorePath, final String athensPublicKey, final String athensPrivateKey) throws Exception {
+//        TrustStore trustStore = new TrustStore(trustStorePath, new JavaKeyStoreProvider(trustStorePath, "secret"));
```
Comment (havetisyan) [inline libs/java/cert_refresher/src/main/java/com/oath/auth/Utils.java]: please remove this commented line since it's identical to the next one.

**Pull request merged**